src/game/Entities/Player.cpp

```diff
@@ -1925,14 +1925,17 @@ bool Player::Mount(uint32 displayid, const Aura* aura/* = nullptr*/)
         return false;
 
     // Custom mount (non-aura such as taxi or command) or in flight: unsummon any pet
-    if (!aura || IsFreeFlying() || IsSpellHaveAura(aura->GetSpellProto(), SPELL_AURA_MOD_FLIGHT_SPEED_MOUNTED))
-        UnsummonPetTemporaryIfAny();
+    if (!aura && !sWorld.getConfig(CONFIG_BOOL_KEEP_PET_WHEN_MOUNTED))
```

Comment:
Pretty sure you broke the bool logic here cos what can happen is you enter this part of the if and you dont do anything. Any flow in this func should either end in "unsummon" or "PET_MODE_DISABLE_ACTIONS"